From 7b244b63d2d1d0eaaece03edf3c13337f46963f4 Mon Sep 17 00:00:00 2001
From: GoProSlowYo <68455785+goproslowyo@users.noreply.github.com>
Date: Sat, 20 May 2023 17:07:43 -0700
Subject: [PATCH] Changes for trumpet and compression

---
 Dockerfile | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/Dockerfile b/Dockerfile
index e7612cc..7cdeca8 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,6 +1,6 @@
 # bump: alpine /FROM alpine:([\d.]+)/ docker:alpine|^3
 # bump: alpine link "Release notes" https://alpinelinux.org/posts/Alpine-$LATEST-released.html
-FROM alpine:3.18.0 AS builder
+FROM alpine:3.18 AS builder
 
 RUN apk add --no-cache \
   coreutils \
@@ -772,6 +772,7 @@ RUN \
   --disable-debug \
   --disable-shared \
   --disable-ffplay \
+  --disable-ffprobe \
   --enable-static \
   --enable-gpl \
   --enable-version3 \
@@ -900,24 +901,27 @@ RUN \
 # make sure binaries has no dependencies, is relro, pie and stack nx
 COPY checkelf /
 RUN \
-  /checkelf /usr/local/bin/ffmpeg && \
-  /checkelf /usr/local/bin/ffprobe
+  /checkelf /usr/local/bin/ffmpeg
+  #/checkelf /usr/local/bin/ffprobe
+
+RUN apk add upx && \
+  upx -v --lzma --best /usr/local/bin/ffmpeg
 
 FROM scratch AS final1
-COPY --from=builder /versions.json /usr/local/bin/ffmpeg /usr/local/bin/ffprobe /
+COPY --from=builder /versions.json /usr/local/bin/ffmpeg /
 COPY --from=builder /usr/local/share/doc/ffmpeg/* /doc/
 COPY --from=builder /etc/ssl/cert.pem /etc/ssl/cert.pem
 
 # sanity tests
 RUN ["/ffmpeg", "-version"]
-RUN ["/ffprobe", "-version"]
+#RUN ["/ffprobe", "-version"]
 RUN ["/ffmpeg", "-hide_banner", "-buildconf"]
 # stack size
 RUN ["/ffmpeg", "-f", "lavfi", "-i", "testsrc", "-c:v", "libsvtav1", "-t", "100ms", "-f", "null", "-"]
 # dns
-RUN ["/ffprobe", "-i", "https://github.com/favicon.ico"]
+#RUN ["/ffprobe", "-i", "https://github.com/favicon.ico"]
 # tls/https certs
-RUN ["/ffprobe", "-tls_verify", "1", "-ca_file", "/etc/ssl/cert.pem", "-i", "https://github.com/favicon.ico"]
+#RUN ["/ffprobe", "-tls_verify", "1", "-ca_file", "/etc/ssl/cert.pem", "-i", "https://github.com/favicon.ico"]
 
 # clamp all files into one layer
 FROM scratch AS final2
-- 
2.39.2

